[GTK][WPE] Add new 'Pointers for contribution' section to manual

nikolaszimmermann opened the pull request:
#### 49b528cf23145a97f8bc9ce1c960d79e5b614ac4
<pre>
[GTK][WPE] Add new &apos;Pointers for contribution&apos; section to manual
<a href="https://bugs.webkit.org/show_bug.cgi?id=293595">https://bugs.webkit.org/show_bug.cgi?id=293595</a>

Reviewed by Adrian Perez de Castro and Michael Catanzaro.

Add new &apos;Pointers for contribution&apos; section to Gtk/WPE manuals.

* Source/WebKit/PlatformGTK.cmake:
* Source/WebKit/PlatformWPE.cmake:
* Source/WebKit/glib/contributing.md.in: Added.
* Source/WebKit/gtk/gtk3-webkitgtk.toml.in:
* Source/WebKit/gtk/gtk4-webkitgtk.toml.in:
* Source/WebKit/wpe/wpewebkit.toml.in:

Canonical link: <a href="https://commits.webkit.org/295867@main">https://commits.webkit.org/295867@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/592c40d8d7ee9f23f08d89e4760fc84f9e9fb51c

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/106380 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/26129 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/16527 "Built successfully") | [  ~~🛠 wpe~~](https://ews-build.webkit.org/#/builders/5/builds/111581 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/56976 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/26796 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/34632 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/5/builds/111581 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/56976 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/109384 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/132/builds/21233 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/96004 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/5/builds/111581 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/133/builds/20725 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/56416 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/90571 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/136/builds/14138 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 gtk~~](https://ews-build.webkit.org/#/builders/2/builds/114440 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/33518 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/24710 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/2/builds/114440 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/33882 "Built successfully") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/8/builds/92233 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/2/builds/114440 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 vision-wk2~~](https://ews-build.webkit.org/#/builders/126/builds/34464 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/137/builds/12283 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/29113 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/17237 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/33443 "Built successfully") | [  ~~🛠 mac-safer-cpp~~](https://ews-build.webkit.org/#/builders/120/builds/38855 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/33189 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/36542 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/34787 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

mcatanzaro commented on Source/WebKit/glib/contributing.md.in:
```diff
@@ -0,0 +1,43 @@
+
+WebKit is a large software project with many contributors and billions of users, receiving a lot of bug reports. Before reporting a new issue, be sure to [search for the existing ones](https://bugs.webkit.org/query.cgi) to make sure your issue may be already known. However, do not refrain from opening a ticket if you cannot find any describing your issue.
+
+If you are unfamiliar with the [WebKit bug writing guidelines](https://bugs.webkit.org/page.cgi?id=bug-writing.html), you might want to have a look before opening a ticket. Prefix the bug title with **[WPE]** or **[GTK]** depending on the port flavour that you are using, and ideally assign a component that matches the problem domain. If unsure, just leave it unassigned.
```
If unsure, select WebKitGTK or WPE WebKit.

The component is the most important part. Putting "[WPE]" or "[GTK]" in the bug report title is nice, but doesn't affect who sees the bug report. Whereas if you select the wrong component, it might be a long time before somebody notices and reassigns the bug to us.

mcatanzaro commented on Source/WebKit/glib/contributing.md.in:
```diff
@@ -0,0 +1,43 @@
+
+To contribute, you should first setup a local development environment. The easiest solution is to use the [WebKit Container SDK](https://github.com/Igalia/webkit-container-sdk). It contains a set of scripts that assist in creating and running a [Podman](https://podman.io/)-based container on your Linux machine, which contains all dependencies needed to compile WebKit.
+
+The **webkit.org** page contains a [section](https://webkit.org/contributing-code/) dedicated to code contributions&mdash;be sure to read it, as it contains many interesting articles, covering the [WebKit Early Warning System (EWS)](http://trac.webkit.org/wiki/EWS), the [coding style guidelines](https://webkit.org/code-style-guidelines/) and much more.
```
Please don't link to trac. If the documentation is important and we don't want to lose it, then it must be migrated to https://docs.webkit.org/.

nikolaszimmermann replied:
Fixed, linked to ews-build.webkit.org

nikolaszimmermann committed:
[GTK][WPE] Add new 'Pointers for contribution' section to manual
https://bugs.webkit.org/show_bug.cgi?id=293595

Reviewed by Adrian Perez de Castro and Michael Catanzaro.

Add new 'Pointers for contribution' section to Gtk/WPE manuals.

* Source/WebKit/PlatformGTK.cmake:
* Source/WebKit/PlatformWPE.cmake:
* Source/WebKit/glib/contributing.md.in: Added.
* Source/WebKit/gtk/gtk3-webkitgtk.toml.in:
* Source/WebKit/gtk/gtk4-webkitgtk.toml.in:
* Source/WebKit/wpe/wpewebkit.toml.in:

Canonical link: https://commits.webkit.org/295867@main
